httpbp/middlewares.go

```diff
@@ -0,0 +1,50 @@
+package httpbp
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+// InjectServerSpan
+func InjectServerSpan(name string, truster HeaderTrustHandler) Middleware {
+	return func(next HandlerFunc) HandlerFunc {
+		return func(ctx context.Context, req *http.Request, resp Response) (body interface{}, err error) {
+			spanHeaders := tracing.Headers{}
+			if truster.TrustSpan(req) {
+				spanHeaders.TraceID = req.Header.Get(TraceIDHeader)
+				spanHeaders.SpanID = req.Header.Get(SpanIDHeader)
+				spanHeaders.Flags = req.Header.Get(SpanFlagsHeader)
+				spanHeaders.Sampled = req.Header.Get(SpanSampledHeader)
+			}
+
+			ctx, span := tracing.StartSpanFromHeaders(ctx, name, spanHeaders)
+			defer func() {
+				span.FinishWithOptions(tracing.FinishOptions{
+					Ctx: ctx,
+					Err: err,
+				}.Convert())
+			}()
+
+			return next(ctx, req, resp)
+		}
+	}
+}
+
+func InjectEdgeRequestContext(truster HeaderTrustHandler, impl *edgecontext.Impl, logger log.Wrapper) Middleware {
+	return func(next HandlerFunc) HandlerFunc {
+		return func(ctx context.Context, req *http.Request, resp Response) (interface{}, error) {
+			var header string
+			if truster.TrustEdgeContext(req) {
+				header = req.Header.Get(EdgeContextHeader)
+			}
+
+			factory := edgecontext.FromRawHeader(header)
+			ctx = edgecontext.InitializeEdgeContext(ctx, impl, logger, factory)
```

Comment:
I think these 2 lines should be inside the if block above, otherwise we'll be spamming the log with errors... Or is the intention to use the error logging inside `InitializeEdgeContext` to log the situation that the http header doesn't work?